src/main/java/bio/terra/workspace/db/DataReferenceDao.java

```diff
@@ -0,0 +1,142 @@
+package bio.terra.workspace.db;
+
+import bio.terra.workspace.app.configuration.WorkspaceManagerJdbcConfiguration;
+import bio.terra.workspace.common.exception.ValidationException;
+import bio.terra.workspace.generated.model.DataReferenceDescription;
+import bio.terra.workspace.generated.model.DataReferenceDescription.CloningInstructionsEnum;
+import bio.terra.workspace.generated.model.DataReferenceDescription.ReferenceTypeEnum;
+import bio.terra.workspace.generated.model.DataReferenceList;
+import bio.terra.workspace.generated.model.ResourceDescription;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DataReferenceDao {
+
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Autowired
+  public DataReferenceDao(WorkspaceManagerJdbcConfiguration jdbcConfiguration) {
+    this.jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+  }
+
+  public DataReferenceList enumerateDataReferences(
```

Comment:
I'm confused about why you are enumerating data references as opposed to enumerating resources.
Data reference is a specialization - one kind of resource. I would have expected an `enumerateResource` that had a filter to only see data references.

There should always be a resource row. If that resource is a reference, then there should also be a data reference row. The general scheme here is fine, but I think it is backwards.